File: urbanairship-sdk/src/main/java/com/urbanairship/messagecenter/AppCompatDelegateWrapper.java
/* Copyright 2017 Urban Airship and Contributors */

package com.urbanairship.messagecenter;

import android.app.Activity;
import android.content.res.Configuration;
import android.os.Bundle;
import android.support.v7.app.ActionBar;
import android.support.v7.app.AppCompatDelegate;
import android.view.MenuInflater;
import android.view.View;
import android.view.ViewGroup;

/**
 * Wrapper for {@link AppCompatDelegate}.
 */
class AppCompatDelegateWrapper {

    private AppCompatDelegate delegate;

    /**
     * Creates an {@code AppCompatDelegateWrapper}.
     *
     * @param activity The activity.
     * @return Instance of {@code AppCompatDelegateWrapper}.
     */
    static AppCompatDelegateWrapper create(Activity activity) {
        AppCompatDelegateWrapper delegateWrapper = new AppCompatDelegateWrapper();
        delegateWrapper.delegate = AppCompatDelegate.create(activity, null);
        return delegateWrapper;
    }

    void onCreate(Bundle savedInstanceState) {
        if (delegate != null) {
            delegate.installViewFactory();
            delegate.onCreate(savedInstanceState);
        }
    }

    void onPostCreate(Bundle savedInstanceState) {
        delegate.onPostCreate(savedInstanceState);
    }

    MenuInflater getMenuInflater() {
        return delegate.getMenuInflater();
    }

    void setContentView(int layoutResId) {
        delegate.setContentView(layoutResId);
    }

    void setContentView(View view) {
        delegate.setContentView(view);
    }

    void setContentView(View view, ViewGroup.LayoutParams params) {
        delegate.setContentView(view, params);
    }

    void addContentView(View view, ViewGroup.LayoutParams params) {
        delegate.addContentView(view, params);
    }

    void onConfigurationChanged(Configuration newConfig) {
        delegate.onConfigurationChanged(newConfig);
    }

    void onPostResume() {
        delegate.onPostResume();
    }

    void onStop() {
        delegate.onStop();
    }

    void invalidateOptionsMenu() {
        delegate.invalidateOptionsMenu();
    }

    void setTitle(CharSequence title) {
        delegate.setTitle(title);
    }

    void onDestroy() {
        delegate.onDestroy();
    }

    ActionBar getSupportActionBar() {
        return delegate.getSupportActionBar();
    }
}

File: urbanairship-core/src/main/java/com/urbanairship/activity/AppCompatDelegateWrapper.kt
/* Copyright Airship and Contributors */
package com.urbanairship.activity

import android.app.Activity
import android.content.res.Configuration
import android.os.Bundle
import android.view.MenuInflater
import android.view.View
import android.view.ViewGroup
import androidx.appcompat.app.ActionBar
import androidx.appcompat.app.AppCompatDelegate

/**
 * Wrapper for [AppCompatDelegate].
 */
internal class AppCompatDelegateWrapper {

    private lateinit var delegate: AppCompatDelegate

    fun onCreate(savedInstanceState: Bundle?) {
        delegate.installViewFactory()
        delegate.onCreate(savedInstanceState)
    }

    fun onPostCreate(savedInstanceState: Bundle?) = delegate.onPostCreate(savedInstanceState)

    val menuInflater: MenuInflater
        get() = delegate.menuInflater

    fun setContentView(layoutResId: Int) = delegate.setContentView(layoutResId)

    fun setContentView(view: View?) = delegate.setContentView(view)

    fun setContentView(view: View?, params: ViewGroup.LayoutParams?) =
        delegate.setContentView(view, params)

    fun addContentView(view: View?, params: ViewGroup.LayoutParams?) =
        delegate.addContentView(view, params)

    fun onConfigurationChanged(newConfig: Configuration?) = delegate.onConfigurationChanged(newConfig)

    fun onPostResume() = delegate.onPostResume()

    fun onStop() = delegate.onStop()

    fun invalidateOptionsMenu() = delegate.invalidateOptionsMenu()

    fun setTitle(title: CharSequence?) = delegate.setTitle(title)

    fun onDestroy() = delegate.onDestroy()

    val supportActionBar: ActionBar?
        get() = delegate.supportActionBar

    companion object {

        /**
         * Creates an `AppCompatDelegateWrapper`.
         *
         * @param activity The activity.
         * @return Instance of `AppCompatDelegateWrapper`.
         */
        fun create(activity: Activity): AppCompatDelegateWrapper {
            val delegateWrapper = AppCompatDelegateWrapper()
            delegateWrapper.delegate = AppCompatDelegate.create(activity, null)
            return delegateWrapper
        }
    }
}
